controllers/profiles/common/properties/application_test.go

```diff
@@ -238,3 +182,140 @@ func doTestGenerateMicroprofileServiceCatalogProperty(t *testing.T, serviceUri s
+	assert.Equal(t, "http://"+platform.Name+"-"+constants.DataIndexServiceName+"."+platform.Namespace+"/processes", generatedProps.GetString(constants.DataIndexServiceURLProperty, ""))
+	assert.Equal(t, "http://"+platform.Name+"-"+constants.JobServiceName+"."+platform.Namespace+"/v2/jobs/events", generatedProps.GetString(constants.JobServiceURLProperty, ""))
+	assert.Equal(t, "false", generatedProps.GetString(constants.JobServiceKafkaSinkInjectionHealthCheck, ""))
+	assert.Equal(t, "", generatedProps.GetString(constants.JobServiceDataSourceReactiveURLProperty, ""))
+	assert.Equal(t, "true", generatedProps.GetString(constants.JobServiceStatusChangeEventsProperty, ""))
+	assert.Equal(t, generatedProps.GetString(constants.JobServiceStatusChangeEventsURL, ""), fmt.Sprintf("%s://%s.%s/jobs", constants.DataIndexServiceURLProtocol, di.GetServiceName(), platform.Namespace))
+
+	// disabling data index bypasses config of outgoing events url
+	platform.Spec.Services.DataIndex.Enabled = nil
+	props, err = NewAppPropertyHandler(workflow, platform)
+	assert.NoError(t, err)
+	generatedProps, propsErr = properties.LoadString(props.WithUserProperties(userProperties).Build())
+	assert.NoError(t, propsErr)
+	assert.Equal(t, 10, len(generatedProps.Keys()))
+	assert.Equal(t, "", generatedProps.GetString(constants.DataIndexServiceURLProperty, ""))
+	assert.Equal(t, "http://"+platform.Name+"-"+constants.JobServiceName+"."+platform.Namespace+"/v2/jobs/events", generatedProps.GetString(constants.JobServiceURLProperty, ""))
+	assert.Equal(t, "false", generatedProps.GetString(constants.JobServiceKafkaSinkInjectionHealthCheck, ""))
+	assert.Equal(t, "", generatedProps.GetString(constants.JobServiceStatusChangeEventsProperty, ""))
+	assert.Equal(t, "", generatedProps.GetString(constants.JobServiceStatusChangeEventsURL, ""))
+
+	// disabling job service bypasses config of outgoing events url
+	platform.Spec.Services.JobService.Enabled = nil
+	props, err = NewAppPropertyHandler(workflow, platform)
+	assert.NoError(t, err)
+	generatedProps, propsErr = properties.LoadString(props.WithUserProperties(userProperties).Build())
+	assert.NoError(t, propsErr)
+	assert.Equal(t, 8, len(generatedProps.Keys()))
+	assert.Equal(t, "", generatedProps.GetString(constants.DataIndexServiceURLProperty, ""))
+	assert.Equal(t, "", generatedProps.GetString(constants.JobServiceURLProperty, ""))
+	assert.Equal(t, "", generatedProps.GetString(constants.JobServiceKafkaSinkInjectionHealthCheck, ""))
+	assert.Equal(t, "", generatedProps.GetString(constants.JobServiceDataSourceReactiveURLProperty, ""))
+	assert.Equal(t, "", generatedProps.GetString(constants.JobServiceStatusChangeEventsProperty, ""))
+	assert.Equal(t, "", generatedProps.GetString(constants.JobServiceStatusChangeEventsURL, ""))
+
+	// check that service app properties are being properly set
+	generatedProps, propsErr = properties.LoadString(NewServiceAppPropertyHandler(platform).WithUserProperties(userProperties).Build())
```

Comment:
in the business code (`services.go`) there is no such case where `NewServiceAppPropertyHandler` is decorated with `WithUserProperties`. Is this a correct population of the test data?
According to what I undertstood from @wmedvede
> there are no DI or JS properties that requires user intervention

so I would expect that there's no way to have `property1`, `property2` in the `generatedProps` (and `kogito.service.url` either), and the len should be 6 instead of 9

Reply:
> there are no DI or JS properties that requires user intervention
while there may not be properties that "require" user intervention... in my opinion, the operator shouldn't be in the business of limiting the user wrt property configs either. we already allow the user the modify just about everything in the DI & JS PodSpec... application properties should be no different. this kind of functionality allows for greater adoption and success in future production-like envs.